badge_server/main.py

```diff
@@ -115,18 +116,20 @@ def _get_self_compatibility_dict(package_name: str) -> dict:
         self incompatibilities. The dict will be formatted like the following:
 
         {
-            'py2': { 'status': 'SUCCESS', 'details': {} },
-            'py3': { 'status': 'SUCCESS', 'details': {} },
+            'py2': { 'status': BadgeStatus.SUCCESS, 'details': {} },
+            'py3': { 'status': BadgeStatus.SUCCESS, 'details': {} },
         }
     """
     pkg = package.Package(package_name)
     compatibility_results = badge_utils.store.get_self_compatibility(pkg)
     result_dict = badge_utils._build_default_result(
-        status='SUCCESS',
+        status=BadgeStatus.SUCCESS,
         details='The package does not support this version of python.')
     for res in compatibility_results:
         pyver = badge_utils.PY_VER_MAPPING[res.python_major_version]
-        result_dict[pyver]['status'] = res.status.value
+        badge_status = PACKAGE_STATUS_TO_BADGE_STATUS.get(
+            res.status) or BadgeStatus.SELF_INCOMPATIBLE
```

Comment:
`res.status` would be one of [these](https://github.com/GoogleCloudPlatform/cloud-opensource-python/blob/5bfba65520f3e4d9dd4f4230337d786f50ef007b/compatibility_lib/compatibility_lib/compatibility_store.py#L36), so we want to convert it to a `BadgeStatus`. Other statuses have a direct equivalent, but in the case of `INSTALL_ERROR` the `BadgeStatus` equivalent would be either `SELF_INCOMPATIBLE` or `INCOMPATIBLE` depending on where it's called. Since this is being called from `_get_self_compatibility_dict()` that's why the default is `SELF_INCOMPATIBLE` here.